node/node.go

```diff
@@ -75,7 +76,7 @@ func New(cfg *Config) (StarknetNode, error) {
 	}
 
 	blockchain := blockchain.NewBlockchain(stateDb, cfg.Network)
-	synchronizer := sync.NewSynchronizer(blockchain, gateway.NewGateway(cfg.Network), log)
+	synchronizer := sync.NewSynchronizer(blockchain, gateway.NewGateway(cfg.Network), clients.NewGatewayClient(cfg.Network.URL()), log)
```

Comment:
L1 verifier should probably be another service. Not embedded inside the synchronizer